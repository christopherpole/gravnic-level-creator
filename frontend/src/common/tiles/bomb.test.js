import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import BombTile from './bomb';

configure({ adapter: new Adapter() });

describe('<BombTile />', () => {
  it('Renders without exploding', () => {
    const bombTile = shallow(<BombTile />);

    expect(bombTile).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const bombTile = shallow(<BombTile />);

    expect(toJson(bombTile)).toMatchSnapshot();
  });

  it('Matches the current snapshot when exploding', () => {
    const bombTile = shallow(<BombTile exploding />);

    expect(toJson(bombTile)).toMatchSnapshot();
  });
});
