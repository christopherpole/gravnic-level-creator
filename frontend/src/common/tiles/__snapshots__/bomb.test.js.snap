// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BombTile /> Matches the current snapshot 1`] = `
<styled.div
  exploding={false}
  moveSpeed={0}
/>
`;

exports[`<BombTile /> Matches the current snapshot when exploding 1`] = `
<styled.div
  exploding={true}
  moveSpeed={0}
/>
`;
